There are a bunch of problems in the CI. The main one is:

```
======================================================================
ERROR: test_async_gen_anext (test.test_asyncgen.AsyncGenAsyncioTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/runner/work/cpython/cpython/Lib/test/test_asyncgen.py", line 401, in test_async_gen_anext
    res = self.loop.run_until_complete(consume())
  File "/Users/runner/work/cpython/cpython/Lib/asyncio/base_events.py", line 641, in run_until_complete
    return future.result()
  File "/Users/runner/work/cpython/cpython/Lib/test/test_asyncgen.py", line 399, in consume
    results.append(await anext(g, 'buckle my shoe'))
TypeError: anext() argument was not async iterable.

----------------------------------------------------------------------
```

and there is also some trailing whitespace. Could you check these?

--- a/Lib/test/test_asyncgen.py
+++ b/Lib/test/test_asyncgen.py
@@ -389,7 +389,7 @@ class AsyncGenAsyncioTest(unittest.TestCase):
                 else:
                     self.yielded += 1
                     return self.yielded
-        
+
         for gen in (agen, MyAsyncIter):
             g = gen()
             async def consume():

--- a/Misc/NEWS.d/next/Core and Builtins/2021-04-07-18-00-05.bpo-43751.8fHsqQ.rst
+++ b/Misc/NEWS.d/next/Core and Builtins/2021-04-07-18-00-05.bpo-43751.8fHsqQ.rst
@@ -1,0 +1,1 @@
+Fixed a bug where ``anext(ait, default)`` would erroneously return None.

--- a/Objects/iterobject.c
+++ b/Objects/iterobject.c
@@ -317,10 +317,41 @@ static PyObject *
 anextawaitable_iternext(anextawaitableobject *obj)
 {
     assert(obj->wrapped != NULL);
-    unaryfunc getter = Py_TYPE(obj->wrapped)->tp_iternext;
+    PyTypeObject *type = Py_TYPE(obj->wrapped);
+    /* Consider the following class:
+     *
+     *     class A:
+     *         async def __anext__(self):
+     *             ...
+     *     a = A()
+     *
+     * Then anext(a) should call
+     * a.__anext__().__await__().__next__()
+     *
+     * On the other hand, given
+     *
+     *     async def agen():
+     *         yield 1
+     *         yield 2
+     *     gen = agen()
+     *
+     * Then anext(g) can just call
+     * g.__anext__().__next__()
+     */
+    if (type->tp_as_async && type->tp_as_async->am_await) {
+        unaryfunc await_getter = type->tp_as_async->am_await;
+        PyObject *result = await_getter(obj->wrapped);
+        if (result == NULL) {
+            return NULL;
+        }
+        type = Py_TYPE(result);
+        Py_SETREF(obj->wrapped, result);
+    }
+
+    unaryfunc getter = type->tp_iternext;
     if (getter == NULL) {
         PyErr_SetString(PyExc_TypeError,
-            "anext() argument was not async iterable.");
+            "__await__() did not return an iterator.");
         return NULL;
     }
     PyObject *result = getter(obj->wrapped);

